xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp

```diff
@@ -349,6 +353,9 @@ bool CRenderManager::Configure()
       return false;
   }
 
+  // configure the renderer to crop if necessary
+  m_pRenderer->ConfigureCropping(m_cropBottom, m_cropTop, m_cropLeft, m_cropRight);
```

Comment:
Well some of those are required for other purposes in Configure but in general yes I believe there are more parts of that method that bare common and could be refactored into the base class. This being my first time in the code I didn't want to refactor the world, I just didn't want to make it any worse. I also wasn't quite sure how to test my changes if I were to refactor every renderer since I can only build and run CWinRenderer.
I also think more smaller functions is more maintainable. 
That said when I change it to a CRect it'll be less of an issue since it'll be one param and one line I'll have to duplicate in every renderer instead of four so if you prefer that I can do that.






On Thu, Mar 24, 2016 at 12:36 AM -0700, "Rainer Hochecker" <notifications@github.com> wrote:












In xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp:
> @@ -349,6 +353,9 @@ bool CRenderManager::Configure()
>        return false;
>    }
>  
> +  // configure the renderer to crop if necessary
> +  m_pRenderer->ConfigureCropping(m_cropBottom, m_cropTop, m_cropLeft, m_cropRight);



the same is true for width, height, etc. following your suggestion we would need extra methods for all of those.



—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub